site/src/pages/types/[entityTypeId].page.tsx

```diff
@@ -0,0 +1,50 @@
+import { GetServerSideProps, NextPage } from "next";
+import { connectToDatabase } from "../../lib/mongodb";
+import { EntityType } from "../../lib/model/entityType.model";
+import { FRONTEND_URL } from "../../lib/config";
+
+const EntityTypeIdPage: NextPage = () => <div />;
+
+export default EntityTypeIdPage;
+
+export const getServerSideProps: GetServerSideProps<
+  {},
+  {
+    entityTypeId: string;
+  }
+> = async (context) => {
+  const { req, res } = context;
+
+  const { db } = await connectToDatabase();
+
+  const entityTypeId = context.query.entityTypeId as string;
+  const entityType = await EntityType.getById(db, { entityTypeId });
+
+  const jsonFlag = new URL(`${FRONTEND_URL}${req.url}`).searchParams.get(
+    "json",
+  );
```

Comment:
```suggestion
  const jsonFlag = Object.prototype.hasOwnProperty.call(context.query, "json");
```

The original statement returns `undefined` for `?json`, which makes the condition below a bit cryptic.

We can replace  `Object.prototype.hasOwnProperty.call` with [`Object.hasOwn`](https://github.com/tc39/proposal-accessible-object-hasownproperty) soon as part of ES2022.

Reply:
Not sure I agree that the proposal is any clearer, but I've added it